zerver/views/__init__.py

```diff
@@ -194,6 +201,12 @@ def accounts_register(request):
             # SSO users don't need no passwords
             password = None
 
+        if create_realm and settings.OPEN_REALM_CREATION:
+            domain = split_email_to_domain(email)
+            do_create_realm(domain, form.cleaned_data['realm_name'])
+            realm = get_realm(domain)
```

Comment:
do_create_realm returns a tuple (realm, created), so you shouldn't need to do a `get_realm` call here by just doing `(realm, _) = do_create_realm(...)`